sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/PartitionKeyInternalHelper.java

```diff
@@ -92,6 +92,32 @@ static public String getEffectivePartitionKeyForHashPartitioningV2(PartitionKeyI
         }
     }
 
+    static String getEffectivePartitionKeyForMultiHashPartitioning(PartitionKeyInternal partitionKeyInternal) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < partitionKeyInternal.components.size(); i++) {
+            try(ByteBufferOutputStream byteArrayBuffer = new ByteBufferOutputStream())  {
+
+                partitionKeyInternal.components.get(i).writeForHashingV2(byteArrayBuffer);
+
+
+                ByteBuffer byteBuffer = byteArrayBuffer.asByteBuffer();
+                UInt128 hashAsUnit128 = MurmurHash3_128.hash128(byteBuffer.array(), byteBuffer.limit());
+
+                byte[] hash = uIntToBytes(hashAsUnit128);
+                Bytes.reverse(hash);
+
+                // Reset 2 most significant bits, as max exclusive value is 'FF'.
+                // Plus one more just in case.
+                hash[0] &= 0x3F;
+
+                stringBuilder.append(HexConvert.bytesToHex(hash));
```

Comment:
Q: How about allocate byte[] and then convert to hex at the end?
Purely from performance wise. .NET micro-benchmark will be useful for quick prototyping.